torchtext/transforms.py

```diff
@@ -189,3 +186,294 @@ def forward(self, input: Any) -> Any:
         :rtype: Union[List[Union[str, int]], List[List[Union[str, int]]]]
         """
         return F.truncate(input, self.max_seq_len)
+
+
+class GPT2BPETokenizer(Module):
+    """
+    Transform for GPT-2 BPE Tokenizer.
+
+    Reimplements openai GPT-2 BPE in TorchScript. Original openai implementation
+    https://github.com/openai/gpt-2/blob/master/src/encoder.py
+
+    :param bpe_encoder: A dict mapping from bpe tokens(e.g sub-words) to
+            bpe token ids. For example: {"ingly": 4420}
+    :type bpe_encoder: Dict[str, int]
+    :param bpe_merge_ranks: A dict mapping a pair of bpe tokens to rank.
+            Torchscript doesn't support tuple as dict key type, concat
+            a pair of string with SEPERATOR (e.g \u0001) as workaround.
+            For example: {"ing\u0001ly": 210}
+    :type bpe_merge_ranks: Dict[str, int]
+    :param byte_encoder: A dict mapping a byte to an unicode character.
+    :type byte_encoder: Dict[int, str]
+    """
+    SEPERATOR: str = "\u0001"
+    LSTRIP_PATTERN: str = "\u0120"
+
+    def __init__(
+        self,
+        encoder_json_path: str,
+        vocab_bpe_path: str,
+    ):
+        super().__init__()
+        self.merge_seperator = self.SEPERATOR
+        self.lstrip_pattern = self.LSTRIP_PATTERN
+        self.bpe_encoder = self._load_bpe_encoder(encoder_json_path)
+        self.bpe_merge_ranks = self._load_bpe_vocab(vocab_bpe_path)
+        self.byte_encoder = bytes_to_unicode()
+
+        self.bpe_decoder = self._load_bpe_decoder(self.bpe_encoder)
+        self.byte_decoder = self._load_byte_decoder(self.byte_encoder)
+        self.inf = len(self.bpe_merge_ranks) + 1
+
+    def _load_bpe_encoder(self, bpe_encoder_path: str) -> Dict[str, int]:
+        with open(get_asset_local_path(bpe_encoder_path), "r", encoding="utf-8") as f:
+            return json.load(f)
+
+    def _load_bpe_vocab(self, bpe_vocab_path: str) -> Dict[str, int]:
+        with open(get_asset_local_path(bpe_vocab_path), "r", encoding="utf-8") as f:
+            bpe_vocab = f.read()
+
+        return {
+            self.merge_seperator.join(merge_pair.split()): i
+            for i, merge_pair in enumerate(bpe_vocab.split("\n")[1:-1])
+        }
```

Comment:
nit: These functions seems to be called internally inside constructor only. can we instead directly implement the logic of them inside constructor instead of making a function call?